src/Badge/Service/ImageFactory.php

```diff
@@ -4,19 +4,24 @@
 
 use App\Badge\Model\BadgeInterface;
 use App\Badge\Model\Image;
+use App\Event\BadgeEvent;
 use PUGX\Poser\Poser;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
 
 /**
  * Class ImageFactory.
  */
 final class ImageFactory
 {
-    public function __construct(private Poser $generator)
+    public function __construct(private Poser $generator, private EventDispatcherInterface $eventDispatcher)
     {
     }
 
     public function createFromBadge(BadgeInterface $badge): Image
     {
+        $badgeEvent = new BadgeEvent($badge);
+        $this->eventDispatcher->dispatch($badgeEvent, BadgeEvent::NAME);
```

Comment:
Not critical, but dispatching events with name is not used anymore. You can remove second argument (and remove constant `NAME` from event, too)

Reply:
Nice catch! Fixed! ;)